easybuild/easyconfigs/m/meshalyzer/meshalyzer-2.0_fix-VTK-linking.patch

```diff
@@ -0,0 +1,15 @@
+fix linking of VTK libraries, avoid that having 'Python' in the installation path messes things up...
+author: Kenneth Hoste (HPC-UGent)
+--- meshalyzer-v2.0/src/Makefile.orig	2020-07-16 08:39:11.000000000 +0200
++++ meshalyzer-v2.0/src/Makefile	2020-09-16 20:38:33.772598607 +0200
+@@ -52,8 +52,8 @@
+ # VTK
+ ifdef VTK_LIBDIR
+ CXXFLAGS      += -DUSE_VTK -I$(VTK_INCDIR)
+-VTK_LSLIBS     = $(shell ls $(VTK_LIBDIR)/libvtk*$(LIB_EXT) | grep -Ev '(Python)|(Java)|(TCL)|(Parallel)|(MPI)|(Filters)|(Imaging)|(Interaction)|(Rendering)|(Chemistry)' )
+-a              = $(subst $(VTK_LIBDIR)/lib,-l,$(VTK_LSLIBS) )
++VTK_LSLIBS     = $(shell ls $(VTK_LIBDIR) | grep libvtk.*$(LIB_EXT) | grep -v $(LIB_EXT).[0-9] | grep -Ev '(Python)|(Java)|(TCL)|(Parallel)|(MPI)|(Filters)|(Imaging)|(Interaction)|(Rendering)|(Chemistry)' )
```

Comment:
my previous suggestion did not actually work, because you have to escape the $ to indicate the end of the line (see below). I tested this and now it works.

I still prefer this because, in your version, if there is a non-library file which happens to contain "so" somewhere in the middle, it will also be included (this is not the case right now, so it's not a big deal). 
 
```suggestion
+VTK_LSLIBS     = $(shell ls $(VTK_LIBDIR) | grep libvtk.*$(LIB_EXT)$$ | grep -Ev '(Python)|(Java)|(TCL)|(Parallel)|(MPI)|(Filters)|(Imaging)|(Interaction)|(Rendering)|(Chemistry)' )
```

Reply:
GitHub is rendering your suggestion a bit strangely, but you just added `$$` after `grep libvtk.*$(LIB_EXT)`, right?

I tried that, didn't work... Here's the patch file I used, am I overlooking something?

```diff
--- meshalyzer-v2.0/src/Makefile.orig	2020-07-16 08:39:11.000000000 +0200
+++ meshalyzer-v2.0/src/Makefile	2020-09-16 20:38:33.772598607 +0200
@@ -52,8 +52,8 @@
 # VTK
 ifdef VTK_LIBDIR
 CXXFLAGS      += -DUSE_VTK -I$(VTK_INCDIR)
-VTK_LSLIBS     = $(shell ls $(VTK_LIBDIR)/libvtk*$(LIB_EXT) | grep -Ev '(Python)|(Java)|(TCL)|(Parallel)|(MPI)|(Filters)|(Imaging)|(Interaction)|(Rendering)|(Chemistry)' )
-a              = $(subst $(VTK_LIBDIR)/lib,-l,$(VTK_LSLIBS) )
+VTK_LSLIBS     = $(shell ls $(VTK_LIBDIR) | grep libvtk.*$(LIB_EXT)$$ | grep -Ev '(Python)|(Java)|(TCL)|(Parallel)|(MPI)|(Filters)|(Imaging)|(Interaction)|(Rendering)|(Chemistry)' )
+a              = $(subst libvtk,-lvtk,$(VTK_LSLIBS) )
 VTK_LIBS       = $(subst $(LIB_EXT),,$(a) )
 LIBS          += $(VTK_LIBS)
 endif
```

Reply:
hm strange, that's exactly the same patch I used.. what's the error you got?

Reply:
Just tried again, I was somehow testing with a wrong copy of the patch... 🤦

Works as expected with updated patch in 671ff866b7